cardano-api/src/Cardano/Api/LedgerState.hs

```diff
@@ -204,12 +216,12 @@ pattern LedgerStateAlonzo st <- LedgerState  (Consensus.LedgerStateAlonzo st)
 
 data FoldBlocksError
   = FoldBlocksInitialLedgerStateError InitialLedgerStateError
-  | FoldBlocksApplyBlockError Text
+  | FoldBlocksApplyBlockError LedgerStateError
 
 renderFoldBlocksError :: FoldBlocksError -> Text
 renderFoldBlocksError fbe = case fbe of
   FoldBlocksInitialLedgerStateError err -> renderInitialLedgerStateError err
-  FoldBlocksApplyBlockError err -> "Failed when applying a block: " <> err
+  FoldBlocksApplyBlockError err -> "Failed when applying a block: " <> textShow err
```

Comment:
What would be better is a `renderLedgerStateError :: LedgerStateError -> Text` function that outputs something similar to the comments you put by the `LedgerStateError` definition.